FhirToDataLake/SynapseValidate/Synapse_Validator.py

```diff
@@ -0,0 +1,207 @@
+            if response.status_code != requests.codes.ok:
+                raise Exception(f"Get data from {url} failed: " + response.text)
+
+            response_content = response.content
+            response_object = json.loads(response_content.decode('utf-8'))
+            if "entry" not in response_object:
+                continuation_token = None
+            else:
+                sumData.extend(response_object["entry"])
+                continuation_token = self.parse_continuation_token(response_object)
+        return sumData
+
+    @staticmethod
+    def parse_continuation_token(response_content):
+        for link_item in response_content["link"]:
+            if link_item["relation"] == "next":
+                parsed_url = urlparse.urlparse(link_item["url"])
+                return urlparse.parse_qs(parsed_url.query)["ct"][0]
+        return None
+
+
+class SchemaManager:
+    def __init__(self, schema_collection_directory):
+        self.schemas = SchemaManager.load_schemas(schema_collection_directory)
+
+    @staticmethod
+    def load_schemas(schema_collection_directory) -> dict:
+        schemas = {}
+        for schema_file in os.listdir(schema_collection_directory):
+            with open(os.path.join(schema_collection_directory, schema_file)) as f:
+                schema = json.load(f)
+                schemas[schema["Type"]] = schema
+        return schemas
+
+    def generate_field_mapping_dicts(self):
+        field_mapping_dicts = {}
+        for resource_type, schema in self.schemas.items():
+            field_mapping_dicts[resource_type] = SchemaManager.generate_external_table_field_mapping_dict(schema)
+        return field_mapping_dicts
+
+    def get_schema(self, resource_type):
+        return self.schemas[resource_type]
+
+    def get_all_resource_types(self):
+        return list(self.schemas.keys())
+
+    @staticmethod
+    def generate_external_table_field_mapping_dict(schema):
+        field_mapping_dict = {}
+        SchemaManager.parse_schema(schema, [], [], field_mapping_dict)
+        return field_mapping_dict
+
+    @staticmethod
+    def parse_schema(node_schema, json_data_path, table_column_path, field_dict_result):
+        if node_schema["IsRepeated"]:
+            field_dict_result[".".join(json_data_path)] = ".".join(table_column_path)
+        elif node_schema["IsLeaf"]:
+            field_dict_result[".".join(json_data_path)] = ".".join(table_column_path)
+        else:
+            for sub_node_name, sub_node_schema in node_schema["SubNodes"].items():
+                table_column_path.append(sub_node_schema["Name"])
+                json_data_path.append(sub_node_schema["Name"])
+
+                SchemaManager.parse_schema(sub_node_schema, json_data_path, table_column_path, field_dict_result)
+
+                table_column_path.pop()
+                json_data_path.pop()
+            return
+
+
+class SQLServerClient:
```

Comment:
Nit: In codes, we only capitalize the first character, just like `Sql`, `Fhir`. In comments, we capitalize them all like `SQL`, `FHIR`. It would be better to keep consistency.